docs/cryptography/playfair-cipher.mdx

```diff
@@ -0,0 +1,503 @@
+---
+title: Playfair Cipher
+slug: /cryptography/playfair-cipher
+---
+
+## Introduction
+
+Playfair cipher is a digraph substitution cipher. By this, we mean that we cipher `2` characters at each stage. We first convert the plaintext given to us into even length format by adding `Z` at the end. We then encrypt the plaintext using a key matrix in pairs of `2`. 
+
+## Explanation
+
+We first check if the given plain text is of even length or not. If found even, we don't require to do anything else we format it by adding a `Z` at the end. Also, we replace `J` in plaintext with `I` if found. Now we generate a `5x5` key matrix which we use to cipher the plaintext. We construct this matrix by putting the alphabets of formatted plaintext in the same order without repetition of alphabets and we fill the later spaces with the remaining alphabets in alphabetical order. We need to note that as we have only 25 spaces for 26 alphabets, generally `J` is omitted. Now we encrypt the plaintext letters pairwise. we take one pair and check for the positions of those letters in the key matrix generated. If we find them in the same row then we pick the right side letter to encrypt(If the letter is at the end we pick the first letter in the row). Similarly, if we find them in the same column then we take the letter below it(If the letter is at the end we pick the first letter in the column). But if both the letters are neither in the same row nor column, then we form a rectangle with these two letters and take the letters on the horizontal opposite corner of the rectangle to encrypt.
+
+Let's have one example: 
+```
+Input : 
+
+Key       :world
+Plaintext :hello
+
+KeyMatrix :
+
+|w |o |r |l |d |
+|--|--|--|--|--|
+|a |b |c |e |f |
+|--|--|--|--|--|
+|g |h |i |k |m |
+|--|--|--|--|--|
+|n |p |q |s |t |
+|--|--|--|--|--|
+|u |v |x |y |z |
+ 
+Formatted plaintext : "he" "ll" "oz"
+Ciphertext          :  kb   ek   dv
+
+Encrypted text will be kbekdv
+```
+
+## Algorithm
+
+//write here in bullet point format or pseudo code
```

Comment:
Put some steps that the reader can follow if they wish to implement them without any reference.